### ISSUE TITLE
Merge issue from PR #119 (parameters)

### ISSUE THREAD

**Issue by felicialim** [AOMediaCodec/iamf#188]
Missing DemixingInfoParamDefinition and ReconGainInfoParamDefinition after merging

**Comment by sunghee-hwang** [comment]
In Audio Element OBU
```
leb128() num_parameters;
for (i = 0; i < num_parameters; i++) {
    leb128() parameter_id;
    leb128() parameter_name;
}
```
[Proposal] 
```
leb128() num_parameters;
for (i = 0; i < num_parameters; i++) {
    leb128() parameter_name;
    ParamDefinition();
}
```

**Comment by sunghee-hwang** [comment]
After reviewing it again, I have realized we need a mapping information between parameter_definition_type and parameter_id.
As parameter_definition() has parameter_id, the mapping information will be between parameter_definition_type and parameter_definition(). And this mapping information should be provided within Descriptor OBUs so that devices can know which parameter block OBU delivers parameter data for the parameter_definition_type.
(But element_mix_config() itself represents its parameter_definition_type, so no need to explicit the type in the config)
Therefore,
 - Table for parameter_definition_type and the class of parameter_definition() need to defined in Audio element OBU or Mix Presentation OBU.
 - parameter_name need to be replaced by parameter_definition_type. 

In Audio Element OBU
```
leb128() num_parameters;
for (i = 0; i < num_parameters; i++) {
    leb128() parameter_defintion_type;
    if (parameter_definition_type == PARAMETER_DEFINITION_DEMIXING_INFO)
        DemixingParamDefinition();
    if (parameter_definition_type == PARAMETER_DEFINITION_RECON_GAIN_INFO)
        ReconGainParamDefinition();
}

class DemixingParamDefintion() extends ParamDefintion() {
}
class ReconGainParamDefintion() extends ParamDefintion() {
}


```

**Comment by sunghee-hwang** [comment]
Also, in element_mix_config, we should let devices know whether parameter block for mix_gain_parameter_data will be present or not.

[As Is]
```
class element_mix_config() {
  MixGainParamDefinition mix_gain;
}
```

[Proposal]
```
class element_mix_config() {
  leb128() mix_gain_type;
  if (mix_gain_type == default)
    signed int (16) default_mix_gain;
  if (mix_gain_type == parameterdata)
    MixGainParamDefinition();
}

class MixGainParamDefinition extends ParamDefinition() {
}
```

**Comment by tdaede** [comment]
Will be fixed by #195 .
